index.js

```diff
@@ -256,6 +258,9 @@ const ContainerService = {
 
     webExtension.startup().then(api => {
       api.browser.runtime.onMessage.addListener((message, sender, sendReply) => {
+        if (message.method === "checkForMajorUpgrade") {
+          sendReply(reason === "upgrade" && MAJOR_VERSIONS.indexOf(self.version) > -1);
```

Comment:
With an embedded WebExtension, the `browser` object available to the legacy add-on code during startup only has the `runtime` property to set up communication between the legacy code and the WebExtension code. (https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Embedded_WebExtensions#Starting_the_WebExtension) And, only the WebExtension code can initiate the communication. 😢 Which is why I had to create another message listener to send a reply to the WebExtension. Only the WebExtension code can mess with the `browserAction` or `extension` APIs.

Or did you mean to move this entire function somewhere else outside of the `startup()` handler?